Lib/idlelib/runscript.py

```diff
@@ -131,22 +138,27 @@ def _run_module_event(self, event):
         if not self.tabnanny(filename):
             return 'break'
         interp = self.shell.interp
-        if pyshell.use_subprocess:
+        if pyshell.use_subprocess and restart:
             interp.restart_subprocess(with_cwd=False, filename=
                         self.editwin._filename_to_unicode(filename))
         dirname = os.path.dirname(filename)
+        argv = [filename]
+        if cli_args:
+            argv += cli_args
         # XXX Too often this discards arguments the user just set...
-        interp.runcommand("""if 1:
+        interp.runcommand(f"""if 1:
             __file__ = {filename!r}
             import sys as _sys
             from os.path import basename as _basename
+            argv = {argv!r}
             if (not _sys.argv or
-                _basename(_sys.argv[0]) != _basename(__file__)):
-                _sys.argv = [__file__]
+                _basename(_sys.argv[0]) != _basename(__file__) or
+                len(argv) > 1):
```

Comment:
Checking the length so that, if there are additional args, they will replace what was already there.  However, existing args won't be replaced by no args.  Currently, F5 still runs with no args and restarts the shell, but that's something to keep in mind if F5 changes.

Reply:
I believe that this can only matter if one used the deprecated -n option.  I have no intention of changing F5